File: LeetCode/1470. Shuffle the Array/solution.cpp
/**
 * 53 / 53 test cases passed.
 * Runtime: 8 ms
 * Memory Usage: 9.6 MB 
 */
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> ans(nums.size());
        for (int i = 0, l = 0, r = n; i < n + n; i += 2, ++ l, ++ r) {
            ans[i] = nums[l];
            ans[i + 1] = nums[r];
        }
        return ans;
    }
};

File: LeetCode/1470. Shuffle the Array/Solution.java
/**
 * 53 / 53 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 41.5 MB 
 */
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int[] ans = new int[nums.length];
        for (int i = 0, j = 0, half = nums.length >> 1; i < half; i++, j += 2) {
            ans[j] = nums[i];
            ans[j + 1] = nums[i + half];
        }
        return ans;
    }
}
